src/main/java/seedu/apollo/storage/Storage.java

```diff
@@ -273,19 +274,39 @@ private static ModuleList readModuleFileContents(File save, Ui ui, ModuleList al
                 if (newModule == null) {
                     throw new InvalidSaveFile();
                 }
+
                 Module module = new Module(newModule.getCode(), newModule.getTitle(), newModule.getModuleCredits());
+                if (isAdded(newModuleList, module)) {
+                    throw new DuplicateModuleInTextFileException();
+                }
                 addLessons(module, newModule, moduleInfoArgs);
                 calendar.addModule(module);
                 newModuleList.add(module);
                 counter++;
             } catch (InvalidSaveFile e) {
                 ui.printInvalidSaveFile(counter, filePath);
+            } catch ( DuplicateModuleInTextFileException e) {
+                ui.printDuplicateModuleInTextFile(counter,moduleDataFilePath);
```

Comment:
Perhaps it would be good to empty module list completely and let the user from scratch as the file is always corrupted.

Reply:
But wouldn't removing everything due to one mistake be inconvenient for the user? also stopping apollo and removing the duplicate and then rerunning solves the issue though. Is that not practical?